### DISCUSSION TITLE
Internationalized Routing

### DISCUSSION THREAD

**Discussion by fedeya** [fedeya/remix-sitemap#29] [Ideas]
This discussion is open for think about how can be the api for help in applications with internationalized routing like `https://example/en/blog` or `https://en.example.com/blog`.

without `map` in each route the `loc` for each language

**Comment by fedeya** [comment]
parhaps to be in accordance with this [definition](https://developers.google.com/search/docs/specialty/international/localized-versions#sitemap) the `alternateRefs` can be used for generate each `url`

**Example**
Turn this
```ts
// routes/about.ts

export const sitemap: SitemapFunction = () => ({
  alternateRefs: [
    {
      href: 'https://example.com',
      hrefLang: 'en',
    },
    {
      href: 'https://es.example.com',
      hrefLang: 'es',
    }
  ]
}) 
```
Into this
```xml
<url>
  <loc>https://example.com/about</loc>
  <xhtml:link
    rel="alternate"
    hreflang="en"
    href="https://example.com/about" 
  />
  <xhtml:link
    rel="alternate"
    hreflang="es"
    href="https://es.example.com/about"
  />
<url>
<url>
  <loc>https://es.example.com/about</loc>
  <xhtml:link
    rel="alternate"
    hreflang="en"
    href="https://example.com/about" 
  />
  <xhtml:link
    rel="alternate"
    hreflang="es"
    href="https://es.example.com/about"
  />
<url>
```

**Comment by prblackburn** [comment]
It would be very useful for our application's usage of this package, if the `alternateRefs` was configurable so that the locale could be in various positions or formats. For instance, we use the `?lang=en` query string rather than `/en/` or `en.domain.com`, but had to patch the module to enable this for our use-case. Could there perhaps be the inclusion of some kind of pattern value to determine how the `href` value is structured with the locale?
